### PR TITLE
Run from the hub build

### PR DESCRIPTION
**Pull request by muffinresearch**
This is a start to moving onto a docker-hub based env.

We can do more things down the road e.g. the npm bits could move into the Dockerfile.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
+  ``docker-compose`` (mac-only).
+
+Creating the docker vm (mac/windows)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you go the docker-machine route your first step is to create a vm::
+
+  docker-machine create --driver=virtualbox addons-dev
+
+Then you can export the variables so that docker-compose  can talk to
```
Comment (wagnerand) [inline docs/index.rst]: Nit: double space.

File: docs/index.rst [hunk 2]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
+the docker service on the vm by running::
+
+This command will tell you what you need to do that::
+
+  docker-machine env addons-dev
```
Comment (wagnerand) [inline docs/index.rst]: Shouldn't this be in line 37?

File: docs/index.rst [hunk 3]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you go the docker-machine route your first step is to create a vm::
+
+  docker-machine create --driver=virtualbox addons-dev
+
+Then you can export the variables so that docker-compose  can talk to
+the docker service on the vm by running::
+
+This command will tell you what you need to do that::
```
Comment (wagnerand) [inline docs/index.rst]: Cross `that`? Or is it just me, not understanding the structure of that sentence?

File: docs/index.rst [hunk 4]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
 
 Please note that any command that would result in files added or modified
 outside of the `olympia` folder (eg modifying pip or npm dependencies) won't be
 persisted, and so won't survive after the container is finished.
-If you need persistence, make sure this command is run in the `Dockerfile` and
-do a full rebuild::
 
-    docker-compose build
+.. note::
+  If you changes to persist those changes should be carried out when the images is
```
Comment (wagnerand) [inline docs/index.rst]: Does not compute. :)

**Comment by wagnerand** [comment]
Awesome job on those docs!

**Comment by wagnerand** [comment]
I will throw away my env (on Mac) and set up a new one following those docs and see if I run into any problems.

File: docs/index.rst (hunk 2 above)
Comment (magopian) [inline docs/index.rst]: This line (and the line 43) don't have enough indentation (hence https://travis-ci.org/mozilla/olympia/jobs/80616630#L203)

File: docs/index.rst [hunk 5]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
 optimizations for security, scalability, L10n, and ease of use, check out
 Mozilla's `Playdoh starter kit <http://playdoh.readthedocs.org/>`_.
 
-
 .. _install-with-docker:
 
 Quickstart
 ----------
+Want the easiest way to start contributing to AMO? Try our docker-based
+development environment.
```
Comment (magopian) [inline docs/index.rst]: Do we want to say that this is the preferred, standard (and only supported?) way to do local dev? ping @andymckay

File: docs/index.rst [hunk 6]
```diff
@@ -154,3 +199,7 @@ Indices and tables
 
 * :ref:`genindex`
 * :ref:`modindex`
+
+
+.. _docker: https://docs.docker.com/installation/#installation
```
Comment (magopian) [inline docs/index.rst]: I like that way of putting all the references to the bottom of the file, maybe we should add some note to the docs (in the contributing?) about that (in another PR ;)

File: docs/index.rst [hunk 7]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
+  Docker recommends installing docker-toolbox_ if you're on OSX or
+  windows and that will provide you with the ``docker-machine`` and
+  ``docker-compose`` (mac-only).
+
+Creating the docker vm (mac/windows)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you go the docker-machine route your first step is to create a vm::
+
+  docker-machine create --driver=virtualbox addons-dev
```
Comment (magopian) [inline docs/index.rst]: I think this also needs more indentation

File: docs/index.rst [hunk 8]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
+
+Then you can export the variables so that docker-compose  can talk to
+the docker service on the vm by running::
+
```
Comment (magopian) [inline docs/index.rst]: Missing the command here

File: docs/index.rst [hunk 9]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
-On the last step, if you're using boot2docker, you-ll need to first find its
-ip, and connect to that instead::
 
-    boot2docker ip
+On the last step, if you're using docker-machine, you-ll
+need to first find its ip, and connect to that instead::
 
-This needs a working installation of docker_, please check the information for
-your operating system.
+    docker-machine ip [name]
```
Comment (magopian) [inline docs/index.rst]: What is `[name]` here? Is it `web`, the name of the "main" container?

File: docs/index.rst [hunk 10]
```diff
@@ -9,55 +9,100 @@ If you want to build a completely different site with all the same Django
...
 
-    docker-compose build  # Installs any new Python requirements in `requirements/*.txt`
-    docker-compose run make update_docker  # Runs database migrations, installs npm modules, rebuilds assets.
+    docker-compose stop
+    docker-compose pull
+    docker-compose up -d
+
+Then from a shell on the running container run::
+
+  make update_docker  # Runs database migrations, installs npm modules, rebuilds assets.
```
Comment (magopian) [inline docs/index.rst]: missing indentation

**Comment by magopian** [comment]
This is very nice, thanks! Looking forward to test that locally. Maybe that will be the tipping point for me to use docker instead of a local install ;)
Would you mind adding `linkcheck` to the travis docs build too?

File: docs/index.rst (hunk 5 above)
Comment (andymckay) [inline docs/index.rst]: I would like suggest we make it the only supported way (other than for what ops do for prod). Having everything run just one way makes it much, much, much easier.

File: docs/index.rst (hunk 9 above)
Comment (wagnerand) [inline docs/index.rst]: `addons` probably.

File: docs/index.rst (hunk 5 above)
Comment (wagnerand) [inline docs/index.rst]: :+1:

File: docs/index.rst (hunk 2 above)
Comment (muffinresearch) [inline docs/index.rst]: Hmm, seems the loop command needs updating it works for me locally.

File: docs/Makefile [hunk 11]
```diff
@@ -28,7 +28,7 @@ help:
 	@echo "  doctest   to run all doctests embedded in the documentation (if enabled)"
 
 loop:
-	python watcher.py 'make html' `find .. -name '*.rst' -o -name '*.py'`
+	python watcher.py 'make html SPHINXOPTS=-nW' `find .. -name '*.rst' -o -name '*.py'`
```
Comment (muffinresearch) [inline docs/Makefile]: Added this, otherwise the test failures due to SPHINXOPTS=-nW aren't picked up locally.

**Comment by muffinresearch** [comment]
> Would you mind adding linkcheck to the travis docs build too?

Let's make that a follow-up.

File: docs/index.rst [hunk 12]
```diff
@@ -9,56 +9,101 @@ If you want to build a completely different site with all the same Django
...
+
+    # On the shell opened by docker-utils:
+    make initialize_docker  # Answer yes, and create your superuser when asked.
     # Once it's all loaded, go to http://localhost:8000 and enjoy!
 
-On the last step, if you're using boot2docker, you-ll need to first find its
-ip, and connect to that instead::
 
-    boot2docker ip
+On the last step, if you're using docker-machine, you-ll
```
Comment (magopian) [inline docs/index.rst]: s/you-ll/you'll/

**Comment by magopian** [comment]
r+wc, thanks for the modifications!

**Pull request merged**